Game/Src/UI/ToggleButton.tscn

```diff
@@ -1,11 +1,37 @@
-[gd_scene load_steps=3 format=2]
+[gd_scene load_steps=5 format=2]
 
 [ext_resource path="res://Src/UI/ToggleButton.gd" type="Script" id=1]
 [ext_resource path="res://Assets/UI/EmptyPwr.png" type="Texture" id=2]
+[ext_resource path="res://Fonts/Enchanted Land.otf" type="DynamicFontData" id=3]
+
+[sub_resource type="DynamicFont" id=1]
+size = 400
+font_data = ExtResource( 3 )
 
 [node name="ToggleButton" type="TextureButton"]
 margin_right = 146.0
 margin_bottom = 132.0
 rect_scale = Vector2( 0.15, 0.15 )
-texture_normal = ExtResource( 2 )
 script = ExtResource( 1 )
+
+[node name="TextureProgress" type="TextureProgress" parent="."]
+margin_right = 40.0
+margin_bottom = 40.0
+max_value = 360.0
+texture_under = ExtResource( 2 )
+fill_mode = 5
+
+[node name="Timer" type="Timer" parent="."]
+
+[node name="Label" type="Label" parent="."]
+anchor_right = 1.0
+anchor_bottom = 1.0
+margin_left = 93.3333
```

Comment:
what is the unit of this? I was expecting pixels but then I guess everything should be int

Reply:
I don't known, this is within godot scenes (not the scripts) so this has been set through the editor. I just added a label that shows the cooldown time